### DISCUSSION TITLE
Nginx randomly dropping connections

### DISCUSSION THREAD

**Discussion by testnetprotonmail** [uNetworking/uWebSockets.js#955] [General]
I *believe* nginx is randomly dropping connections. It seems very random but roughly 50% of the time my requests for opening a socket never make it to the nodeJS instance, (I log the req in the upgradeHandler as a means to verify if it was received)

Every time this happens a log is created in error.log of nginx:

```connect() failed (111: Unknown error) while connecting to upstream,```

Which seems to indicate the error lies with the uWebsockets.js instance, and there my trail ends as my uWebsockets.js app never seems to be relayed the request.

```
const app = uWS.App({

})


app.ws('/ws/*', {

  /* Handlers */
  upgrade: (res, req, context) => { console.log(...incoming...)})})
```
& then my nginx right now is just 

```
location /ws {
    proxy_pass  url;
    proxy_http_version 1.1;
}
```

I've tried many different nginx configurations, all kinds of headers, but couldn't figure it out.
I've switched from a different socket library with the same nginx configurations and never had this.

Is there perhaps any way to attain additional logs from the Uwebsockets app (maybe there is something to log before the upgrade handler that would clue me in on the solution?

**Comment by ezioda004** [comment]
I believe you need to let nginx know how to proxy certain headers, example setup could be like:
```nginx
http {
    map $http_upgrade $connection_upgrade {
        default upgrade;
        '' close;
    }
 
    upstream websocket {
        server 192.168.100.10:8010;
    }
 
    server {
        listen 8020;
        location / {
            proxy_pass http://websocket;
            proxy_http_version 1.1;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection $connection_upgrade;
            proxy_set_header Host $host;
        }
    }
}
```
Reference: https://www.nginx.com/blog/websocket-nginx/

**Comment by testnetprotonmail** [comment]
Disabling ipv6 on the host fixed it.

https://serverfault.com/questions/317393/connect-failed-111-connection-refused-while-connecting-to-upstream
